File: src/metreos/core/io/DataHandler.java
// $Id$
//
// Created by wert on Apr 8, 2007.
//
// Copyright (c) 2007 Cisco Systems, Inc. All rights reserved.

package metreos.core.io;

import metreos.core.Who;
import metreos.util.FlexBuffer;

/**
 * Interface used to deliver data from a stream source.
 */
public interface DataHandler extends SourceHandler<DataSource>
{
	/**
	 * Delivers data from a stream source.
	 * @param sender
	 * @param buf
	 * @throws Exception
	 */
	public void data( Who sender, FlexBuffer buf ) throws Exception;
}

File: binding-ruby/src/main/ruby/transport/data_handler.rb
module DataHandler
  
  #Interface used to deliver data from a stream source.
  #
  # Delivers data from a stream source.
  # @param src
  # @param sender
  # @param buf
  # @throws Exception
  #
   
  def data( src, sender, buf )
    raise "subclass responsibility"
  end
  
end